tests/modules.rs

```diff
@@ -771,6 +771,88 @@ fn doc_attribute_on_module() {
     .run();
 }
 
+#[test]
+fn group_attribute_on_module() {
```

Comment:
We should have a test for the ordering of submodules in groups if `--unsorted` is used.

My first thought was that submodules should be ordered based on their position in the `justfile`. However, I'm now not sure if that's correct. I think it's natural to have submodules all declared together at the top or bottom of a `justfile`, and natural to list submodules at the end of each group.

So maybe my comment above is off, and we should, instead of using `submodule.name.unwrap().offset` in the ordering key, use `usize::MAX` instead.

Reply:
Hm, I tested setting `submodule.name.unwrap().offset` vs. `usize::MAX`, and both have the same behavior. Both implementations output the submodules at the end of a group because of how we output recipes and then submodules in `subcommand.rs`:

```rust
for recipe in recipes {...}
for (i, submodule) in submodules.iter().enumerate() {...}
```

If we wanted the ability to order based on the submodule's position in the `justfile`, then we would need to interleave the recipes and submodules. But since it sounds like want them at the end of a group, I'll go with `usize::MAX`. Let me know if you disagree.

Reply:
I added tests for `--unsorted` with and without `--list-submodules`. I ended up going with `submodule.name.unwrap().offset` because I think that better guarantees the submodules are ordered how they appear in the `justfile` at the bottom of a group.